addons/event/models/event_event.py

```diff
@@ -705,6 +705,14 @@ def mail_attendees(self, template_id, force_send=False, filter_func=lambda self:
             for attendee in event.registration_ids.filtered(filter_func):
                 self.env['mail.template'].browse(template_id).send_mail(attendee.id, force_send=force_send)
 
+    def _get_date_begin_event_tz(self):
```

Comment:
I find it a bit odd that we have to do this work manually. Can't you set it inside the context or something of the sort?

From what I can see, the date widget used in the pdf report templates with the "date_only" option also uses "babel.dates.format_date", exactly like "format_date" does (and also not caring about "tz_name"), yet it works in the reports and not in the mail templates?

Reply:
Hello @awa-odoo 

All changes are done. I must still check something for lna.

Regarding timezone, indeed, we can use the datetime widget, it seems to work just fine. I've added the missing tz_name parameter, that was missing, my bad.

About the _set_tz_context() in the report templates: I've added it on the event for badges to harmonize the file but I'm not sure it is useful in practice. It is not sufficient a least, we still need to give tz_name.

What do you think?

Reply:
Hello :wave: 

Looks like the datetime widget does not fully take into account the timezone from the context and needs the extra param to make it work properly.

I would still keep the "_set_tz_context" call for safety though, in case it's used in the template somehow and because it makes more sense compared to what we do elsewhere in the functional event flows.

Thank you for cleaning it!

Let me know when it's ready, for me we can launch it.
(If you modify things, let me know what, so I don't have to re-read everything ^^).

Would be nice to launch it asap as we need the upgrade as well, and the freeze is "officially" today :disguised_face:

Reply:
Ok, I've just added a small commit regarding translation and date format in reporting. I give a shot to luc's small remaining remarks and it should be ok :)